recipes-core/date-sync/init

```diff
@@ -1,42 +1,76 @@
 #!/bin/sh
-#
 ### BEGIN INIT INFO
-# Provides:		date-sync
-# Required-Start:	$remote_fs $syslog $networking
-# Required-Stop:	$remote_fs $syslog
-# Default-Start:	2 3 4 5
-# Default-Stop:		1
-# Short-Description:	Start and stop the date-sync daemon
+# Provides:             date-sync
+# Required-Start:       $remote_fs $syslog $networking
+# Required-Stop:        $remote_fs $syslog
+# Default-Start:        2 3 4 5
+# Default-Stop:         1
+# Short-Description:    Start and stop the date-sync daemon
 ### END INIT INFO
-#
 
 PATH=/usr/local/sbin:/usr/local/bin:/sbin:/bin:/usr/sbin:/usr/bin
 NAME=date-sync
 DESC="Date Sync"
 LOGFILE=/tmp/date-sync.log
+PIDFILE=/var/run/date-sync.pid
 
-start() {
-	echo -n "Starting $DESC: "
+# Configuration
+NTP_SERVER="pool.ntp.org"
+SYNC_INTERVAL=60  # Run every 60 seconds (1 minute)
+MAX_LOG_SIZE=524288  # 512KB (0.5MB) in bytes
 
-	# update time by fetching the latest one from ntp
-	ntpd -q -p pool.ntp.org
+log() {
+    echo "$(date +"%Y-%m-%d %H:%M:%S"): $1" >> $LOGFILE
+    if [ -f "$LOGFILE" ] && [ $(stat -c %s "$LOGFILE") -gt $MAX_LOG_SIZE ]; then
+        tail -n 1000 "$LOGFILE" > "${LOGFILE}.tmp" && mv "${LOGFILE}.tmp" "$LOGFILE"
```

Comment:
That would only copy the current logfile to store it for historical logs. And would only do it once.
What I am doing here is rotating the file such that it trims it regularly to keep its size under the max log size.

The reason behind this is because currently "for some reason" we are storing the logs under /tmp. Which is in memory (tmpfs). 
This is actually not good for the long term and I am planning later to refactor all our init scripts (e.g. reth, rbuilder, prometheus, etc...) to send their logs to /encrpted_disk/(SERVICE_NAME).log instead.

Reply:
what difference would this make compared to the current approach?